### PR TITLE
Enabled test_matmul_v2_op for final state Eager Dygraph

### PR DESCRIPTION
**Pull request by jim19930609**
### PR types
New features

### PR changes
Others

### Describe
Enabled test_matmul_v2_op for final state Eager Dygraph

### CODE REVIEW

**Review by wanghuancoder — DISMISSED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/test_matmul_v2_op.py [hunk 1]
```diff
@@ -1,64 +1,56 @@
...
+    if (not Out.shape):
+        Out = np.array([Out], dtype='float64')
     return Out
 
 
 class TestMatMulV2Op(OpTest):
-    """
-    case 1
-    """
+    '\n    case 1\n    '
```
Comment (zhiqiu) [inline python/paddle/fluid/tests/unittests/test_matmul_v2_op.py]: why change this?

File: python/paddle/fluid/tests/unittests/test_matmul_v2_op.py [hunk 2]
```diff
@@ -401,148 +350,152 @@ def test_dygraph_fp16(self):
...
+        self.check_output(check_eager=True)
 
     def test_check_grad_normal(self):
         self.check_grad(
             ['X', 'Y'],
             'Out',
             user_defined_grads=[self.grad_x, self.grad_y],
-            user_defined_grad_outputs=[self.grad_out])
+            user_defined_grad_outputs=[self.grad_out],
+            check_eager=False)
```
Comment (zhiqiu) [inline python/paddle/fluid/tests/unittests/test_matmul_v2_op.py:461]: why check_eager is false here？

Comment (jim19930609) [reply python/paddle/fluid/tests/unittests/test_matmul_v2_op.py:461]: We haven't supported Type Promotion at Dygraph level yet, so I turned of eager mode test for "test_check_grad_normal" and "test_check_grad_ignore_y" for now.

File: python/paddle/fluid/tests/unittests/test_matmul_v2_op.py (hunk 1 above)
Comment (jim19930609) [reply python/paddle/fluid/tests/unittests/test_matmul_v2_op.py]: Hmmm, it's a change from astunparse, let's sync offline.

**Review by zhiqiu — APPROVED** [review]
LGTM

**Pull request merged**